src/test/suite/connectionController.test.ts

```diff
@@ -915,4 +924,328 @@ suite('Connection Controller Test Suite', function () {
+        sshTunnelPort: 22
+      }
+    };
+    const mockMigratePreviouslySavedConnection: any = sinon.fake.resolves({
+      id: '1d700f37-ba57-4568-9552-0ea23effea89',
+      name: 'localhost:27017',
+      storageLocation: 'GLOBAL',
+      connectionOptions: {
+        connectionString: 'mongodb://localhost:27017/?readPreference=primary&ssl=false'
+      }
+    });
+
+    sinon.replace(
+      testConnectionController,
+      '_migratePreviouslySavedConnection',
+      mockMigratePreviouslySavedConnection
+    );
+
+    await testConnectionController._getConnectionInfoWithSecrets(oldSavedConnectionInfo);
+
+    assert(
+      mockMigratePreviouslySavedConnection.called === true,
+      'Expected _migratePreviouslySavedConnection to be called.'
+    );
+  });
+
+  test('_getConnectionInfoWithSecrets does not run a migration for new connections', async () => {
+    const connectionInfo = {
+      id: '1d700f37-ba57-4568-9552-0ea23effea89',
+      name: 'localhost:27017',
+      storageLocation: StorageLocation.GLOBAL,
+      connectionOptions: {
+        connectionString: 'mongodb://localhost:27017/?readPreference=primary&ssl=false'
+      }
+    };
+    await testConnectionController._storageController.saveConnectionToStore(connectionInfo);
+    await testConnectionController.loadSavedConnections();
+
+    const connections = testConnectionController.getSavedConnections();
+
+    assert(
+      connections.length === 1,
+      `Expected connections to be 1 found ${connections.length}`
+    );
+
+    const mockMigratePreviouslySavedConnection: any = sinon.fake();
+
+    sinon.replace(
+      testConnectionController,
+      '_migratePreviouslySavedConnection',
+      mockMigratePreviouslySavedConnection
+    );
+
+    const newSavedConnectionInfoWithSecrets = await testConnectionController._getConnectionInfoWithSecrets(connections[0]);
+
+    assert.deepStrictEqual(
+      newSavedConnectionInfoWithSecrets,
+      connectionInfo
+    );
+    assert(
+      mockMigratePreviouslySavedConnection.called === false,
+      'Expected _migratePreviouslySavedConnection to not be called.'
+    );
+  });
+
+  test('addNewConnectionStringAndConnect saves connection without secrets to the global store', async () => {
+    const mockConnect: any = sinon.fake.resolves({ successfullyConnected: true });
+
+    sinon.replace(
+      testConnectionController,
+      '_connect',
+      mockConnect
+    );
+
+    await vscode.workspace
+      .getConfiguration('mdb.connectionSaving')
+      .update('defaultConnectionSavingLocation', DefaultSavingLocations.Global);
+    await testConnectionController.addNewConnectionStringAndConnect(
+      TEST_DATABASE_URI_USER
+    );
+
+    const workspaceStoreConnections = testConnectionController._storageController.get(
+      StorageVariables.GLOBAL_SAVED_CONNECTIONS
+    );
+
+    assert(
+      workspaceStoreConnections,
+      `Expected workspace store has onnections found ${workspaceStoreConnections}`
```

Comment:
```suggestion
      `Expected workspace store has connections found ${workspaceStoreConnections}`
```

or

```suggestion
      `Expected workspace store to have connections, found ${workspaceStoreConnections}`
```

(or just drop the message altogether, it doesn’t really provide much value here anyway)